internal/pkg/daemon/bpfrecorder/bpfrecorder.go

```diff
@@ -394,11 +596,20 @@ func (b *BpfRecorder) Load(startEventProcessor bool) (err error) {
 		return fmt.Errorf("architecture %s is currently unsupported", runtime.GOARCH)
 	}
 
-	module, err := b.NewModuleFromBufferArgs(&bpf.NewModuleArgs{
-		BPFObjBuff: bpfObject,
-		BPFObjName: "recorder.bpf.o",
-		BTFObjPath: b.btfPath,
-	})
+	if b.recordingMode == recordingSeccomp {
```

Comment:
I am wondering if we should move this code behind an interface instead of using `recordingMode` to fork the code execution. 

You can define an interface which is implemented for both use cases seccomp and apparmor. This would make the code more clear.

Reply:
I modified the code to introduce an interface for the two mentioned points as a start.